lib/transfer.c

```diff
@@ -918,13 +918,42 @@ static CURLcode readwrite_upload(struct Curl_easy *data,
 
   do {
     curl_off_t nbody;
+    ssize_t offset = 0;
+
+    if(0 != k->upload_present && k->upload_present < 1024 && !k->upload_done) {
+      DEBUGASSERT(data->set.upload_buffer_size >= 1024);
+      offset = k->upload_present;
+#ifndef CURL_DISABLE_HTTP
+      /* disable appending to upload buffer if HTTP body not ready */
+      if(conn->handler->protocol & PROTO_FAMILY_HTTP) {
+        if(conn->bits.authneg)
+          offset = 0;
+#ifdef USE_HYPER
+        offset = 0;
+#else
```

Comment:
This function is actually never called for HTTP when Hyper is used. The HTTP route is different then.

Reply:
ok.  will remove that along with next update with review feedback.

Reply:
done